src/shared/lib/markDownRules.tsx

```diff
@@ -141,15 +225,39 @@ const markDownRules: RenderRules = {
     );
   },
   image: node => {
-    return (
-      <>
-        <CachedImage
+    const src = node.attributes?.src;
+    const mimeType = mime.lookup(src) || '';
+
+    if (mimeType.startsWith('audio/')) {
+      return <AudioPlayer uri={src} title={node.content} key={node.key} />;
+    } else if (mimeType.startsWith('video/') || src?.includes('.quicktime')) {
+      return <VideoPlayer uri={src} key={node.key} />;
+    } else if (src?.includes('youtu')) {
+      const videoId = src.split('.be/')?.[1];
+      const uri = src?.includes('watch?')
+        ? src
+        : `https://www.youtube.com/embed/${videoId}`;
+
+      return (
+        <WebView
+          width="100%"
+          height={250}
           key={node.key}
-          resizeMode="contain"
-          style={localStyles.image}
-          source={node.attributes.src}
+          mediaPlaybackRequiresUserAction
+          source={{
+            uri,
+          }}
         />
-      </>
+      );
+    }
```

Comment:
Please move the logic to YoutubeVideo component so that you can use here like this:
```
else if (src?.includes('youtu')) {
  return <YoutubeVideo uri={src} key={node.key} />
}
```